### ISSUE TITLE
`migrate dev` fails on second run with specific schema

### ISSUE THREAD

**Issue by janpio** [prisma/prisma#10983]
1. Connect to a MySQL 8 database server (not sure if specific to that, but only recreated with that) and create the following tables:
	  ```sql
	  CREATE TABLE `product_item` (
	    `id` int(11) NOT NULL AUTO_INCREMENT,
	    `uuid` varchar(30) NOT NULL,
	    `cost` decimal(15,2) NOT NULL DEFAULT '0.00',
	    `createdAt` datetime(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6),
	    `updatedAt` datetime(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6),
	    `deletedAt` datetime(6) DEFAULT NULL,
	    `lastUserId` varchar(60) DEFAULT NULL,
	    `status` varchar(20) NOT NULL DEFAULT '',
	    `statusNoteId` int(11) DEFAULT NULL,
	    `sequenceNumber` int(11) DEFAULT NULL,
	    `scheduledDate` date DEFAULT NULL,
	    `componentId` int(11) NOT NULL,
	    PRIMARY KEY (`id`),
	    UNIQUE KEY `IDX_2fc7a6ec8c1e67f13f8cf9bffb` (`uuid`),
	    UNIQUE KEY `REL_ccdae40f3258a45d4883d22f6c` (`statusNoteId`),
	    UNIQUE KEY `IDX_ccdae40f3258a45d4883d22f6c` (`statusNoteId`),
	    UNIQUE KEY `componentId` (`componentId`,`sequenceNumber`),
	    KEY `product-item-uuid-idx` (`uuid`),
	    KEY `product-item-status-idx` (`status`),
	    KEY `last-user-id-idx` (`lastUserId`),
	    KEY `IDX_SCHEDULEDDATE` (`scheduledDate`)
	  );
	  
	  CREATE TABLE `attachment` (
	    `id` int(11) NOT NULL AUTO_INCREMENT,
	    `createdAt` datetime(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6),
	    `updatedAt` datetime(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6) ON UPDATE CURRENT_TIMESTAMP(6),
	    `deletedAt` datetime(6) DEFAULT NULL,
	    `lastUserId` varchar(60) DEFAULT NULL,
	    `fileName` varchar(500) NOT NULL,
	    `note` text,
	    `path` varchar(1023) NOT NULL,
	    `type` varchar(255) NOT NULL,
	    `productItemId` int(11) NOT NULL,
	    PRIMARY KEY (`id`),
	    KEY `FK_d5749de348b0f64483a3f020046` (`productItemId`),
	    KEY `IDX_fileName` (`fileName`),
	    KEY `IDX_deletedAt` (`deletedAt`),
	    CONSTRAINT `FK_d5749de348b0f64483a3f020046` FOREIGN KEY (`productItemId`) REFERENCES `product_item` (`id`)
	  );
	  ```
2. In a new folder, run the following:
	  ```
	  npm init -y
	  npm install prisma @prisma/client
	  npx prisma init --datasource-provider mysql
	  ```
3. Add your connection string to `.env`
4.  Add `previewFeatures = ["extendedIndexes"]` to `generator client` in your schema file
5. Run `npx prisma db pull`
6. Check that the schema file is nicely populated with this schema:
	  ```prisma
	  generator client {
	    provider        = "prisma-client-js"
	    previewFeatures = ["extendedIndexes"]
	  }
	  
	  datasource db {
	    provider = "mysql"
	    url      = env("DATABASE_URL")
	  }
	  
	  model attachment {
	    id            Int          @id @default(autoincrement())
	    createdAt     DateTime     @default(now()) @db.DateTime(6)
	    updatedAt     DateTime     @default(now()) @db.DateTime(6)
	    deletedAt     DateTime?    @db.DateTime(6)
	    lastUserId    String?      @db.VarChar(60)
	    fileName      String       @db.VarChar(500)
	    note          String?      @db.Text
	    path          String       @db.VarChar(1023)
	    type          String       @db.VarChar(255)
	    productItemId Int
	    product_item  product_item @relation(fields: [productItemId], references: [id], onDelete: NoAction, onUpdate: NoAction, map: "FK_d5749de348b0f64483a3f020046")
	  
	    @@index([productItemId], map: "FK_d5749de348b0f64483a3f020046")
	    @@index([deletedAt], map: "IDX_deletedAt")
	    @@index([fileName], map: "IDX_fileName")
	  }
	  
	  model product_item {
	    id             Int          @id @default(autoincrement())
	    uuid           String       @unique(map: "IDX_2fc7a6ec8c1e67f13f8cf9bffb") @db.VarChar(30)
	    cost           Decimal      @default(0.00) @db.Decimal(15, 2)
	    createdAt      DateTime     @default(now()) @db.DateTime(6)
	    updatedAt      DateTime     @default(now()) @db.DateTime(6)
	    deletedAt      DateTime?    @db.DateTime(6)
	    lastUserId     String?      @db.VarChar(60)
	    status         String       @default("") @db.VarChar(20)
	    statusNoteId   Int?         @unique(map: "IDX_ccdae40f3258a45d4883d22f6c")
	    sequenceNumber Int?
	    scheduledDate  DateTime?    @db.Date
	    componentId    Int
	    attachment     attachment[]
	  
	    @@unique([componentId, sequenceNumber], map: "componentId")
	    @@index([scheduledDate], map: "IDX_SCHEDULEDDATE")
	    @@index([lastUserId], map: "last-user-id-idx")
	    @@index([status], map: "product-item-status-idx")
	    @@index([uuid], map: "product-item-uuid-idx")
	  }
	  ```
7. Run `npx prisma migrate dev` and get this:
	  ```
	  C:\Users\Jan\Documents\throwaway\10405_2>npx prisma migrate dev
	  Environment variables loaded from .env
	  Prisma schema loaded from prisma\schema.prisma
	  Datasource "db": MySQL database "10405_2" at "tiny-tests.cm0mkpwj8arx.eu-central-1.rds.amazonaws.com:3306"
	  
	  - Drift detected: Your database schema is not in sync with your migration history.
	  
	  The following is a summary of the differences between the expected database schema given your migrations files, and the actual schema of the database.  
	  
	  It should be understood as the set of changes to get from the expected schema to the actual schema.
	  
	  [+] Added tables
	    - attachment
	    - product_item
	  
	  [*] Changed the `attachment` table
	    [+] Added foreign key on columns (productItemId)
	  
	  - The following migration(s) are applied to the database but missing from the local migrations directory: 20220106105014_init
	  
	  √ We need to reset the MySQL database "10405_2" at "tiny-tests.cm0mkpwj8arx.eu-central-1.rds.amazonaws.com:3306".
	  Do you want to continue? All data will be lost. ... yes
	  
	  √ Enter a name for the new migration: ... first
	  Applying migration `20220106105306_first`
	  
	  The following migration(s) have been created and applied from new schema changes:
	  
	  migrations/
	    └─ 20220106105306_first/
	      └─ migration.sql
	  
	  Your database is now in sync with your schema.
	  
	  ✔ Generated Prisma Client (3.7.0 | library) to .\node_modules\@prisma\client in 108ms
	  ```
8. Run `npx prisma migrate dev` again and get this:
	  ```
	  C:\Users\Jan\Documents\throwaway\10405_2>npx prisma migrate dev
	  Environment variables loaded from .env
	  Prisma schema loaded from prisma\schema.prisma
	  Datasource "db": MySQL database "10405_2" at "tiny-tests.cm0mkpwj8arx.eu-central-1.rds.amazonaws.com:3306"
	  
	  
	  ⚠️  Warnings for the current datasource:
	  
	    • A unique constraint covering the columns `[uuid]` on the table `product_item` will be added. If there are existing duplicate values, this will fail.  • A unique constraint covering the columns `[statusNoteId]` on the table `product_item` will be added. If there are existing duplicate values, this will fail.
	    • A unique constraint covering the columns `[componentId,sequenceNumber]` on the table `product_item` will be added. If there are existing duplicate values, this will fail.
	  
	  √ Are you sure you want create and apply this migration? ... yes
	  √ Enter a name for the new migration: ... second
	  Applying migration `20220106105325_second`
	  Error: P3018
	  
	  A migration failed to apply. New migrations cannot be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve
	  
	  Migration name: 20220106105325_second
	  
	  Database error code: 1061
	  
	  Database error:
	  Duplicate key name 'FK_d5749de348b0f64483a3f020046'
	  
	  Please check the query number 10 from the migration file.
	  ```

**Comment by pimeys** [comment]
More minimal example:

```prisma
datasource db {
  provider = "mysql"
  url      = "mysql://root:prisma@localhost:3307/prisma"
}

generator js {
  provider        = "prisma-client-js"
  previewFeatures = ["extendedIndexes"]
}

model attachment {
  id        Int       @id @default(autoincrement())
  deletedAt DateTime? @db.DateTime(6)

  @@index([deletedAt], map: "IDX_deletedAt")
}
```
```sql
CREATE TABLE `attachment` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `deletedAt` datetime(6) DEFAULT NULL,
  PRIMARY KEY (`id`),
  KEY `IDX_deletedAt` (`deletedAt`)
)
```

**Comment by janpio** [comment]
This change got merged in `3.8.0-dev.47` and can now be tested.

I could already confirm the minimal example is fixed. Moving on to the original one now.
